src/sortedvector.jl

```diff
@@ -0,0 +1,86 @@
+"""
+    SortedVector
+
+A wrapper for sorted vectors, designed for fast unions.
+
+# Constructor
+
+    SortedVector(data::AbstractVector; already_sorted=false)
+
+# Example
+
+```jldoctest
+x = SortedVector([3, 4, 2])
+x = SortedVector([1, 3, 5]; already_sorted=true)
+z = union(x, y)
+
+# output
+
+SortedVector([1, 2, 3, 4, 5])
+````
+"""
+struct SortedVector{T,V<:AbstractVector{T}} <: AbstractVector{T}
+    data::V
+
+    function SortedVector{T,V}(data::V; already_sorted=false) where {T,V<:AbstractVector{T}}
+        if already_sorted
+            new{T,V}(data)
+        else
+            new{T,V}(sort(data))
+        end
+    end
+
+    function SortedVector{T,V}(x::Number) where {T,V<:AbstractVector{T}}
+        return new{T,V}(convert(V, [T(x)]))
+    end
+
+    function SortedVector{T,V}() where {T,V<:AbstractVector{T}}
+        return new{T,V}(convert(V, T[]))
+    end
```

Comment:
could this be improved by referencing a constant empty `SortedVector` like you did with other types?

Reply:
Couldn't we expect `new{T,V}(V())` to work?

Reply:
you're right!